src/sqlfluff/dialects/dialect_vertica.py

```diff
@@ -0,0 +1,2036 @@
+    # TODO: add other math operators
+    [
+        StringLexer("integer_division", "//", CodeSegment),
+    ],
+    before="divide",
+)
+
+# Set Keywords
+vertica_dialect.update_keywords_set_from_multiline_string(
+    "unreserved_keywords", vertica_unreserved_keywords
+)
+
+vertica_dialect.update_keywords_set_from_multiline_string(
+    "reserved_keywords", vertica_reserved_keywords
+)
+vertica_dialect.sets("reserved_keywords").difference_update(["ROWS"])
+
+vertica_dialect.sets("bare_functions").clear()
+vertica_dialect.sets("bare_functions").update(
+    ["CURRENT_TIMESTAMP", "CURRENT_TIME", "CURRENT_DATE", "LOCALTIME", "LOCALTIMESTAMP", "SYSDATE"]
+)
+
+# Add all Vertica encoding types
+vertica_dialect.sets("encoding_types").clear()
+vertica_dialect.sets("encoding_types").update(
+    [
+        "AUTO",
+        "BLOCK_DICT",
+        "BLOCKDICT_COMP",
+        "BZIP_COMP",
+        "COMMONDELTA_COMP",
+        "DELTARANGE_COMP",
+        "DELTAVAL",
+        "GCDDELTA",
+        "GZIP_COMP",
+        "RLE",
+        "ZSTD_COMP",
+        "ZSTD_FAST_COMP",
+        "ZSTD_HIGH_COMP"
+    ],
+)
+
+# Add all Vertica compression types (for COPY statement)
+vertica_dialect.sets("compression_types").clear()
+vertica_dialect.sets("compression_types").update(
+    [
+        "UNCOMPRESSED",
+        "BZIP",
+        "GZIP",
+        "LZO",
+        "ZSTD",
+    ],
+)
+
+vertica_dialect.sets("date_part_function_name").update(
+    ["DATEADD", "DATEDIFF", "EXTRACT", "DATE_PART"]
+)
+
+# Add datetime units
+# https://docs.vertica.com/latest/en/sql-reference/functions/data-type-specific-functions/datetime-functions/date-part/
+vertica_dialect.sets("datetime_units").update(
+    [
+        # millennium
+        "MILLENNIUM",
+        # century
+        "CENTURY",
+        # decade
+        "DECADE",
+        # epoch
+        "EPOCH",
+        # year
+        "YEAR",
+        "ISOYEAR",
+        # quarter
+        "QUARTER",
+        # month
+        "MONTH",
+        # week
+        "WEEK",
+        "ISOWEEK",
+        # day of week
+        "ISODOW",
+        "DOW",
+        # day of year
+        "DOY",
+        # day
+        "DAY",
+        # hour
+        "HOUR",
+        # minute
+        "MINUTE",
+        # second
+        "SECOND",
+        # millisec
+        "MILLISECONDS",
+        # microsec
+        "MICROSECONDS",
+        # timezone
+        "TIME ZONE",
+        "TIMEZONE_HOUR",
+        "TIMEZONE_MINUTE",
+    ]
+)
+
+vertica_dialect.add(
+    EncodingType=OneOf(
+        MultiStringParser(
+            vertica_dialect.sets("encoding_types"),
+            KeywordSegment,
+            type="encoding_type",
+        ),
+        MultiStringParser(
+            [
+                f"'{encoding}'"
+                for encoding in vertica_dialect.sets("encoding_types")
+            ],
+            KeywordSegment,
+            type="encoding_type",
+        ),
```

Comment:
Been a while since I looked at this, but confused by this seems to be processing the `encoding_types` set twice?